sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -46,7 +49,12 @@ type Controller struct {
 	timeout     time.Duration
 	clock       controllerTime.Clock
 	ticker      controllerTime.Ticker
-	notifier    *dynamicconfig.Notifier
+	notifier    *notify.Notifier
+	schedules *[]*pb.ConfigResponse_MetricConfig_Schedule
+	// Timestamp all metrics with CollectionPeriod were last exported
+	lastCollected map[pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod]time.Time
+	// Maps instrument name to a CollectionPeriod
```

Comment:
One thing we can maybe consider down the line is to have a common set of in-memory models from which we can marshal and unmarshal the protobuf structs. This way, we can keep error-checking and validation to the models, plus add any additional useful functionality (and the names would be less verbose, too).  I have some already in extension/dynamicconfig/model on the collector contrib repo, but we could consider moving them to a more common location, if this sounds like a good idea?

Reply:
I agree this is something to consider down the line, but maybe not for us. Protobuf types are used throughout the SDK, and I think this is something the open telemetry maintainers are thinking about https://github.com/open-telemetry/opentelemetry-go/issues/793. They say they are thinking about building a pipeline, and I worry any work we do on this front would be made redundant if our code is merged someday.